packages/babel-parser/src/plugins/flow.js

```diff
@@ -2823,4 +2861,416 @@ export default (superClass: Class<Parser>): Class<Parser> =>
+      );
+    }
+
+    flowEnumErrorDuplicateMemberName(
+      pos: number,
+      { enumName, memberName }: { enumName: string, memberName: string },
+    ): void {
+      this.raise(
+        pos,
+        `Enum member names need to be unique, but the name \`${memberName}\` has already been used ` +
+          `before in enum \`${enumName}\`.`,
+      );
+    }
+
+    flowEnumErrorInconsistentMemberValues(
+      pos: number,
+      { enumName }: { enumName: string },
+    ): void {
+      this.raise(
+        pos,
+        `Enum \`${enumName}\` has inconsistent member initializers. Either use no initializers, or ` +
+          `consistently use literals (either booleans, numbers, or strings) for all member initializers.`,
+      );
+    }
+
+    flowEnumErrorInvalidExplicitType(
+      pos: number,
+      {
+        enumName,
+        suppliedType,
+      }: { enumName: string, suppliedType: null | string },
+    ): void {
+      const suggestion =
+        `Use one of \`boolean\`, \`number\`, \`string\`, or \`symbol\` in ` +
+        `enum \`${enumName}\`.`;
+      const message =
+        suppliedType === null
+          ? `Supplied enum type is not valid. ${suggestion}`
+          : `Enum type \`${suppliedType}\` is not valid. ${suggestion}`;
+      this.raise(pos, message);
+    }
+
+    flowEnumErrorInvalidMemberInitializer(
+      pos: number,
+      { enumName, explicitType, memberName }: EnumContext,
+    ): void {
+      let message = null;
+      switch (explicitType) {
+        case "boolean":
+        case "number":
+        case "string":
+          message =
+            `Enum \`${enumName}\` has type \`${explicitType}\`, so the initializer of ` +
+            `\`${memberName}\` needs to be a ${explicitType} literal.`;
+          break;
+        case "symbol":
+          message =
+            `Symbol enum members cannot be initialized. Use \`${memberName},\` in ` +
+            `enum \`${enumName}\`.`;
+          break;
+        default:
+          // null
+          message =
+            `The enum member initializer for \`${memberName}\` needs to be a literal (either ` +
+            `a boolean, number, or string) in enum \`${enumName}\`.`;
+      }
+      this.raise(pos, message);
+    }
+
+    flowEnumErrorNumberMemberNotInitialized(
+      pos: number,
+      { enumName, memberName }: { enumName: string, memberName: string },
+    ): void {
+      this.raise(
+        pos,
+        `Number enum members need to be initialized, e.g. \`${memberName} = 1\` in enum \`${enumName}\`.`,
+      );
+    }
+
+    flowEnumErrorStringMemberInconsistentlyInitailized(
+      pos: number,
+      { enumName }: { enumName: string },
+    ): void {
+      this.raise(
+        pos,
+        `String enum members need to consistently either all use initializers, or use no initializers, ` +
+          `in enum \`${enumName}\`.`,
+      );
+    }
+
+    flowEnumMemberInit(): EnumMemberInit {
+      const lookahead = this.lookahead();
```

Comment:
Lookahead is really expensive, and should be avoided if possible.
Can we first parse the literal, and then check if the new token is `comma || braceR`?

Reply:
Will address